src/diem/utils.py

```diff
@@ -172,3 +179,11 @@ def balance(account: jsonrpc.Account, currency: str) -> int:
         if b.currency == currency:
             return b.amount
     return 0
+
+
+def to_snake(o: typing.Any) -> str:  # pyre-ignore
+    if isinstance(o, str):
+        return "".join(["_" + i.lower() if i.isupper() else i for i in o]).lstrip("_")
+    elif hasattr(o, "__name__"):
+        return to_snake(getattr(o, "__name__"))
+    return to_snake(type(o))
```

Comment:
do we want `'JSONParser' => 'j_s_o_n_parser'` or `json_parser`?

Reply:
It will end with j_s_o_n_parser.
Don't think I want it, but keep logic simple is better here, hence I will name it JsonParser if I have to create one.
This func is for creating snake case name from a model class, and there are only couple classes, hence simple version is good enough